### PR TITLE
Fix scope for FSpatialNetBitReader lifecycle

### PR DESCRIPTION
**Pull request by ImprobableNic**
#### Description
There can only be a single FSpatialNetBitReader in a given stack. Their scope in ApplyRPCInternal was not small enough and a complex scenario while calling RPCs could loop the NetDriver back  in the stack.

#### Tests

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp [hunk 1]
```diff
@@ -418,13 +418,15 @@ FRPCErrorInfo SpatialRPCService::ApplyRPCInternal(UObject* TargetObject, UFuncti
 	FMemory::Memzero(Parms, Function->ParmsSize);
 
 	TSet<FUnrealObjectRef> UnresolvedRefs;
-	TSet<FUnrealObjectRef> MappedRefs;
-	RPCPayload PayloadCopy = PendingRPCParams.Payload;
-	FSpatialNetBitReader PayloadReader(NetDriver->PackageMap, PayloadCopy.PayloadData.GetData(), PayloadCopy.CountDataBits(), MappedRefs,
-									   UnresolvedRefs);
+	{
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp:421]: Extra scoping, love this pattern : )

File: SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp [hunk 2]
```diff
@@ -418,13 +418,15 @@ FRPCErrorInfo SpatialRPCService::ApplyRPCInternal(UObject* TargetObject, UFuncti
...
+		TSet<FUnrealObjectRef> MappedRefs;
+		RPCPayload PayloadCopy = PendingRPCParams.Payload;
+		FSpatialNetBitReader PayloadReader(NetDriver->PackageMap, PayloadCopy.PayloadData.GetData(), PayloadCopy.CountDataBits(), MappedRefs,
```
Comment (simonsarginson) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp:424]: are we a 100% sure NetDriver can't be null?

Comment (simonsarginson) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/RPCs/SpatialRPCService.cpp:424]: ah ok legacy code, I see now the diff is just adding scope

**Review by simonsarginson — APPROVED** [review]
LGTM

**Review by m-samiec — APPROVED** [review]

**Review by joshuahuburn — APPROVED** [review]

**Pull request merged**